frankenstein/chunk_store.go

```diff
@@ -199,15 +209,20 @@ func bigBuckets(from, through model.Time) []int64 {
 }
 
 // Put implements ChunkStore
-func (c *AWSChunkStore) Put(chunks []wire.Chunk) error {
+func (c *AWSChunkStore) Put(ctx context.Context, chunks []wire.Chunk) error {
+	userID, err := userID(ctx)
+	if err != nil {
+		return err
+	}
+
 	// TODO: parallelise
 	for _, chunk := range chunks {
 		err := timeRequest("Put", s3RequestDuration, func() error {
 			var err error
 			_, err = c.s3.PutObject(&s3.PutObjectInput{
 				Body:   bytes.NewReader(chunk.Data),
 				Bucket: aws.String(c.bucketName),
-				Key:    aws.String(chunk.ID),
+				Key:    aws.String(fmt.Sprintf("%s/%s", userID, chunk.ID)),
```

Comment:
We have `fmt.Sprintf("%s/%s", userID, chunk.ID)` four times now in the code - time for a helper function?